include/fiction/algorithms/simulation/sidb/occupation_probability_of_excited_states.hpp

```diff
@@ -29,21 +29,25 @@ namespace fiction
  * @param temperature System temperature to assume.
  * @return The occupation probability of all erroneous states is returned.
  */
-[[nodiscard]] inline double occupation_probability_gate_based(const energy_and_state_type& energy_distribution,
-                                                              const double                 temperature) noexcept
+[[nodiscard]] inline double occupation_probability_gate_based(const sidb_energy_and_state_type& energy_distribution,
```

Comment:
So this function receives an argument called `energy_distribution`, which is however not of type `sidb_energy_distribution`... you lost me here :joy:

Reply:
You are right! This is more than confusing!